testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,294 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	logrunner "github.com/leehinman/spigot/pkg/runner"
+	"github.com/sajari/regression"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/go-sysinfo"
+	"github.com/elastic/go-sysinfo/types"
+	"github.com/elastic/go-ucfg"
+)
+
+type ExtendedRunner struct {
+	suite.Suite
+	info         *define.Info
+	agentFixture *atesting.Fixture
+
+	ESHost string
+}
+
+type ComponentMetrics struct {
+	Memory        MemoryMetrics  `mapstructure:"memstats"`
+	Handles       HandlesMetrics `mapstructure:"handles"`
+	UnixTimestamp int64
+}
+
+// TestComponent is used as a key in our map of component metrics
+type TestComponent struct {
+	Binary   string `mapstructure:"binary"`
+	Dataset  string `mapstructure:"dataset"`
+	ID       string `mapstructure:"id"`
+	CompType string `mapstructure:"type"`
+}
+
+type MemoryMetrics struct {
+	GcNext      uint64 `mapstructure:"gc_next"`
+	MemoryAlloc uint64 `mapstructure:"memory_alloc"`
+	MemorySys   uint64 `mapstructure:"memory_sys"`
+	MemoryTotal uint64 `mapstructure:"memory_total"`
+	RSS         uint64 `mapstructure:"rss"`
+}
+
+type HandlesMetrics struct {
+	Open  int           `mapstructure:"open"`
+	Limit HandlesLimits `mapstructure:"limit"`
+}
+
+type HandlesLimits struct {
+	Hard uint   `mapstructure:"hard"`
+	Soft uint64 `mapstructure:"soft"`
+}
+
+// MetricsSystem is used for windows handles metrics
+type MetricsSystem struct {
+	Handles HandlesMetrics `mapstructure:"handles"`
+}
+
+type processWatcher struct {
+	handle     types.Process
+	pid        int
+	name       string
+	regHandles *regression.Regression
+}
+
+func TestLongRunningAgentForLeaks(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{Type: define.Linux},
+			{Type: define.Windows},
+		},
```

Comment:
Looking at [the test report](https://buildkiteartifacts.com/e0f3970e-3a75-4621-919f-e6c773e2bb12/0187faac-2a90-4642-bc78-c33cefa167ba/018da45a-3876-46f8-a97d-acf0e1f37a71/018da45b-11ae-4d52-8a74-68c2b5b29e3d/build/TEST-report.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LSFQJXY5T%2F20240215%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240215T212811Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjENP%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJIMEYCIQCHh1pok0KTIJemTXKBWBlrD8xxPwwKMAxKfGhjDlhHIQIhAKt4V%2F5exE9cD6vyBQ8Ho%2FwGgZtU9MiYZGgM8bT6QT32KvoDCKv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMDMyMzc5NzA1MzAzIgyDsUHT6Km%2F4n8XZoMqzgMC22NOmLqlh0XL051C9OPPrZbCzy%2BbR8LgV7sQD0%2FEPpNFg9jZQTGnk3BrxoEZZPSlhl4FAF193m9L7T3DQ0OOjL49aMXBn4mMcP%2F40l%2BUfsuaA59RxE0MP3X4RzKTNcK0PQi%2FLlu%2BaUKEwpgmkHG3%2BDiKUQgxmF5cMtsxSzEyE881d2uThKT3RsOElnerKx3mmfilE8dyvRA1ft2NIRJkLlRTJLHLlCfDBxeCUe4uRMz%2BA4QxS5BpYyHrXNwk6LGgCVQC4elL3vhBK8yGm0yQr4gJZD83EKsxVhuZNx3N7GTJ5BpL7mIScBCaLyFHBkxIB7X%2BHEPZRPA1pFqwT3POKx1N4g94QpDdrIfhIS7uokgLu0eF%2BMw1Kuba73PtkxZk2hSAxe6kSnWG5nbA%2BDzrWdSbwOHSmXwKmN1IBD7oHh2ac1qCVdSVCFvNQDHg8FztIUjDUds%2FAEV483%2Fpr%2BKD4uGcS5tcHDE%2BFJ6uRyoH388SCU2DFpLEET7q482iijkBBfmRU15FUYGQPGbWNfpdxAj%2BJRJ08iMSs4o3aqsp6SbTLcqdKXuLU5WYpNps22qTnmloenvOrdmBNHZ7X8rqrs9xTq%2FO51ibo7jK4aMwj6a5rgY6pAGfSdtSlnAbvI%2FeFdWkMIt1syup0vaphDGYZ6XptW3ejeEuSmLoIl4aRyQKYUq5OxtJdJnJZnS%2FKE59QJSA51cytpIeksZYoJO5PvWhuDXFdJTq80YigHcHDAm29UCdQ9mww6y7Mws2pKz4TvQLH1Nj5bxcfgBJZnH%2BI53vcK5lCOnR3DCR327aSfhMLc%2FjL08cPpGc4yZRPJrwVfOKLZdIUWRLqw%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=3f4160d5987aab4161702b9e0fce5345a07450c9a7ccf02e42f03dbd7f1710ae) for the most recent run in https://buildkite.com/elastic/elastic-agent/builds/7148#018da45b-11ae-4d52-8a74-68c2b5b29e3d it doesn't look like it ran on Windows?

Reply:
So, I can see the windows output in the latest CI run, unless I wildly misunderstanding it: 
```
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): go: downloading github.com/billgraziano/dpapi v0.4.0
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): go: downloading github.com/jmespath/go-jmespath v0.4.0
```

However, more worryingly, the test fails due to errors with spigot, but this doesn't actually fail the CI run: 

```
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): # github.com/leehinman/spigot/pkg/output/syslog
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): ..\go\pkg\mod\github.com\leehinman\spigot@v0.0.0-20230530144131-c5a179785123\pkg\output\syslog\syslog.go:45:24: undefined: syslog.Dial
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): ..\go\pkg\mod\github.com\leehinman\spigot@v0.0.0-20230530144131-c5a179785123\pkg\output\syslog\syslog.go:65:59: undefined: syslog.Priority
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): ..\go\pkg\mod\github.com\leehinman\spigot@v0.0.0-20230530144131-c5a179785123\pkg\output\syslog\syslog.go:66:18: undefined: syslog.Priority
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): ..\go\pkg\mod\github.com\leehinman\spigot@v0.0.0-20230530144131-c5a179785123\pkg\output\syslog\syslog.go:70:14: undefined: syslog.LOG_EMERG
>>> (windows-amd64-2022-fleet) Test output (sudo) (stderr): ..\go\pkg\mod\github.com\leehinman\spigot@v0.0.0-20230530144131-c5a179785123\pkg\output\syslog\syslog.go:72:14: undefined: syslog.LOG_ALERT
```
cc @leehinman

Reply:
That a build failure in the tests doesn't fail the integration test step definitely seems like a bug in the runner we should fix, separate from whatever is happening with spigot.

Syslog doesn't exist on Windows, it looks like the `log/syslog` package spigot imports from the standard library has build tags on it to exclude it from Windows. https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/log/syslog/syslog.go;l=5